pkg/models/profilescanrequest.go

```diff
@@ -0,0 +1,13 @@
+//
+// Copyright (C) 2024 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package models
+
+// ProfileScanRequest is the struct for requesting a profile for a specified device.
+type ProfileScanRequest struct {
+	DeviceName  string `json:"deviceName" validate:"required"`
+	ProfileName string `json:"profileName,omitempty"`
+	Options     any    `json:"options,omitempty"`
```

Comment:
add `protocols map[string]models.ProtocolProperties`
SDK could retrieve the protocol properties from cache and pass to the driver, so the driver doesn't need to query the protocol properties on its own